### PR TITLE
Feature/disable chat filter when spam disabled

### PR DESCRIPTION
**Pull request by rcalderoni**
changed it to not apply any filtering/muting unless SPAM is turned ON, and also refactored to move the 'educate user about claims/trapped' code into own function so it is no longer nested inside mute check code

the logic seems to flow because:

- the config check for SPAM already causes mute check to return false
- all but the final of the if/else blocks were related to spam checking

note that the final if/else block which has been moved outside of spam check entirely: code for social logging and ignore check is below spam config check, and will happen regardless if spam feature is enabled, but not if spam is caught (to conform with the previous if/else style flow)

### CODE REVIEW

**Comment by RoboMWM** [comment]
> Ok I made a few changes and a PR, it builds but needs review and a second pair of eyes before testing, you may find something wrong with it that I didn't consider, for example we may want to alert the admin that if they attempt to softmute when SPAM feature is disabled it won't work (which in my opinion should be a feature of SPAM only anyways, doesn't make sense to me that GP without SPAM just has a mute feature thrown in, especially considering there are other plugins that handle this and we don't want them to conflict).

So I don't know how many have disabled spam yet use these features, especially since [25% disable spam](https://bstats.org/plugin/bukkit/GriefPrevention-legacy#uses_spam)

It's also highly advised to do one PR per feature/refactor, as the diff becomes more unreadable the more changes you make.

**Review by RoboMWM — CHANGES_REQUESTED** [review]
My time is sporadic for the next month or so. If anyone else can review this that'd help, or if you can split this into separate PRs, because the diff most consists of you moving code around - so it's very difficult to see what you actually changed.

If you can, split up the PRs (or if one depends on another, just submit the first one), one for each change, so it's easier to see the diff and merge each change.

**Comment by RoboMWM** [comment]
Ok, took a look at the first commit of this PR and I see why the diff is so hard to follow now. You only did refactoring (some of which is questionable at best, like instantiating a HashSet when it may never be used on every single chat message). Then I assume later on you made the primary changes that this PR is supposed to do.

For PRs, it's a lot easier if you just submit one at a time. Either the refactor first, or the feature change first. (I'd say the feature change first, then later you can PR refactored names for better clarity.) By mixing them the overall diff becomes almost useless to read on its own.

In other words, PR just the feature change. You can refactor other parts of the class in other, separate PRs.
